### PR TITLE
Smaller tables

### PR DESCRIPTION
**Pull request by jimrandomh**
After lots of whack-a-mole indexing, especially on the Posts table, it occurred to me that we should instead (or also) be optimizing to make table scans reasonably fast. Currently rows of the Posts table are *extremely* bloated: first by fields that are outright unused (migrations/renames which failed to drop them), again by `legacyData` (which for old-LW contains an extra copy of post bodies), and again by `contents` (which is denormalized from the `Revisions` table).

I picked a random old sequences post, to see how much it could be shrunk. Sizes measured in bytes of pretty-printed JSON.
 * Initial size: 67520.
Remove legacyData, htmlBody, body, commenters, content, htmlHighlight, upvoters (denormalized fields that should have already been removed, but weren't)
 * Reduced size: 21225 (3.2x reduction)
Removing contents (denormalized from Revisions table)
 * Reduced size: 2267 (30x reduction)
Removing whitespace from the JSON
 * Reduced size: 1683 (40x reduction)

Given that a table scan of the Revisions table takes ~1s and our threshold for acceptability is ~100ms, this should make table scans acceptably performant.

This PR:
 * Drops a bunch of unused fields which were left over from old versions, and bloating the collections
 * Moves `legacyData` to its own collection (off of `Posts`, `Comments` and `Users`)
 * Adds a script which looks at `contents` (and other `makeEditable` fields) to confirm that they're really just denormalized, and don't contain unique content

After verifying that `contents` contains no unique data, we will want to follow up with a second PR which makes resolvers use the Revisions table rather than using the denormalized field, and drops the denormalized field.

### CODE REVIEW

**Comment by jimrandomh** [comment]
Ran this on the dev DB. Initially, table scans (`db.posts.find({nonexistentFieldName:"foo"}).explain("executionStats")`) took about 1.0s. After running the `dropObsoleteColumns` migration, that fell to 0.6s. I'm currently waiting for the `moveLegacyData` migration to run, but I believe the results of `dropObsoleteColumns` are sufficient to validate the model that table-scanning has a cost proportional to the size of the data.

**Comment by jimrandomh** [comment]
With both migration scripts run, it's about 0.4s. That leaves `contents`, which is a bit more involved to remove, but probably takes it down to <0.2s and maybe even <0.1s.

**Comment by jimrandomh** [comment]
I don't think this is waiting on anything except review.

File: packages/lesswrong/lib/collections/legacyData/collection.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { createCollection } from 'meteor/vulcan:core';
+import { addUniversalFields } from '../../collectionUtils'
+import { ensureIndex } from '../../collectionUtils';
+
+const schema = {
+  _id: {
+    type: String
+  },
+  objectId: {
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/legacyData/collection.js]: We've used the variable name `documentId` throughout the rest of the codebase, so I think we should use that instead

Comment (jimrandomh) [reply packages/lesswrong/lib/collections/legacyData/collection.js]: Vulcan has sort of poisoned that name, though, with its hacks in the Connectors; lessening the scope of impact of that hack seems more important than any particular name.

**Pull request merged**